staging/src/k8s.io/client-go/transport/round_trippers.go

```diff
@@ -87,6 +93,17 @@ type authProxyRoundTripper struct {
 	rt http.RoundTripper
 }
 
+// NewCacheRoundTripper creates a roundtripper that reads the ETag on
+// response headers and send the If-None-Match header on subsequent
+// corresponding requests.
+func NewCacheRoundTripper(cacheDir string, rt http.RoundTripper) http.RoundTripper {
+	d := diskv.New(diskv.Options{BasePath: cacheDir})
+	t := httpcache.NewTransport(diskcache.NewWithDiskv(d))
```

Comment:
@apelisse 
>>Looks like when If-Modified-Since is used (which the httpcache library does), the server >>should return an HTTP 304

>Yes,

>>(which this PR's implementation should preserve)

>It doesn't

Hrm. I found what looks like a  [relevant spec section](https://tools.ietf.org/html/rfc7234#section-4.3.2) but I don't have time to think about it too closely right now.

The [primary use case](https://github.com/kubernetes/kubernetes/issues/49966#issuecomment-319510084) @caesarxuchao and I were discussing to make use of this feature depends on the caller's ability to determine whether the response is cached. I'm hoping somebody more knowledgable can help me understand whether we have misguided expectations or whether httpcache is misbehaving.